Code/Engine/Core/Utils/Blackboard.h

```diff
@@ -64,18 +64,28 @@ EZ_DECLARE_REFLECTABLE_TYPE(EZ_CORE_DLL, ezBlackboardEntryFlags);
 /// and then NPCs might use that information to make decisions.
 class EZ_CORE_DLL ezBlackboard : public ezRefCounted
 {
-private:
+public:
   ezBlackboard();
+  ~ezBlackboard();
 
-public:
-  /// \brief Factory method to create a new blackboard.
+  /// \brief Factory method to get access to a globally registered blackboard.
   ///
-  /// Since blackboards use shared ownership we need to make sure that blackboards are created in ezCore.dll.
-  /// Some compilers (MSVC) create local v-tables which can become stale if a blackboard was registered as global but the dll
-  /// which created the blackboard is already unloaded.
-  static ezSharedPtr<ezBlackboard> Create(ezAllocatorBase* pAllocator = ezFoundation::GetDefaultAllocator());
-  ~ezBlackboard();
+  /// If a blackboard with that name was already created globally before, its reference is returned.
+  /// Otherwise it will be created and permanently registered under that name.
+  /// Global blackboards cannot be removed. Although you can change their name via "SetName()",
+  /// the name under which they are registered globally will not change.
+  ///
+  /// If at some point you want to "remove" a global blackboard, instead call UnregisterAllEntries() to
+  /// clear all its values.
+  static ezSharedPtr<ezBlackboard> GetOrCreateGlobal(const ezHashedString& sBlackboardName, ezAllocatorBase* pAllocator = ezFoundation::GetDefaultAllocator());
+
+  /// \brief Finds a global blackboard with the given name.
+  static ezSharedPtr<ezBlackboard> FindGlobal(const ezTempHashedString& sBlackboardName);
 
+  /// \brief Changes the name of the blackboard.
```

Comment:
I would remove the SetName method to avoid confusion. The name needs to be defined when the factory method is called.

Reply:
I had to keep it, because it wouldn't work with components during editing.